### PR TITLE
handle gql conflict gracefully

### PR DESCRIPTION
**Pull request by maorfr**
intercept the `409: Conflict` returned from https://github.com/app-sre/qontract-server/blob/master/src/server.ts#L56 and exit gracefully with exit code `3`, which will indicate this conflict.

this will make the output much clearer to our users (think of fluentd).

### CODE REVIEW

**Comment by jmelis** [comment]
the fluentd sidecar I'm working on will have conflicts with this change. Can we hold this after I submit the fluentd sidecar PR?

**Comment by maorfr** [comment]
sure. i did it so the fluentd setup would be simpler, but as you wish.

File: reconcile/cli.py [hunk 1]
```diff
@@ -164,6 +165,14 @@ def run_integration(func, *args):
     except RunnerException as e:
         sys.stderr.write(str(e) + "\n")
         sys.exit(1)
+    except GqlApiError as e:
+        if '409' in str(e):
```
Comment (jmelis) [inline reconcile/cli.py]: a bit fragile, can we detect the exit code a bit more robustly?

File: utils/gql.py [hunk 2]
```diff
@@ -30,7 +32,11 @@ def __init__(self, url, token=None):
 
     def query(self, query, variables=None):
         try:
-            result_json = self.client.execute(query, variables)
+            # supress print on HTTP error
```
Comment (jmelis) [inline utils/gql.py]: why do we want to suppress the message?

File: reconcile/cli.py (hunk 1 above)
Comment (maorfr) [reply reconcile/cli.py]: agreed to improve the GqlApiError in the future

Comment (jmelis) [reply reconcile/cli.py]: add a `TODO` and let's leave this for the future.

File: utils/gql.py (hunk 2 above)
Comment (maorfr) [reply utils/gql.py]: agreed to keep as is

**Pull request merged**